ocaml/xapi/nm.ml

```diff
@@ -331,27 +330,19 @@ let bring_pif_up ~__context ?(management_interface=false) (pif: API.ref_PIF) =
 
         (* If the PIF is a bond master, the bond slaves will now go down *)
         (* Interface-reconfigure in bridge mode requires us to set currently_attached to false here *)
-        begin match rc.API.pIF_bond_master_of with
-          | [] -> ()
+        let rc = match rc.API.pIF_bond_master_of with
+          | [] -> rc
           | bond :: _ ->
-            let dbBond = Db.Bond.get_record ~__context ~self:bond in
-
-            if dbBond.API.bond_auto_update_mac then begin
+            let bond_record = Db.Bond.get_record ~__context ~self:bond in
+            List.iter (fun self -> Db.PIF.set_currently_attached ~__context ~self ~value:false) bond_record.API.bond_slaves;
+            if bond_record.API.bond_auto_update_mac
+            then
               (* Make sure the MAC address of the PIF Bond master matches the MAC address of the Bond's primary slave *)
-              let primarySlaveRef = dbBond.API.bond_primary_slave in
-              let primarySlaveMAC = Db.PIF.get_MAC ~__context ~self:primarySlaveRef in
-              if rc.pIF_MAC <> primarySlaveMAC then begin
-                Db.PIF.set_MAC ~__context ~self:pif ~value:primarySlaveMAC;
-                mac_changed := true;
-              end;
-            end;
-
-            let slaves = dbBond.API.bond_slaves in
-
-            List.iter (fun self -> Db.PIF.set_currently_attached ~__context ~self ~value:false) slaves
-        end;
-
-        let rc = if !mac_changed then Db.PIF.get_record ~__context ~self:pif else rc in
+              let primary_slave_ref = bond_record.API.bond_primary_slave in
+              let primary_slave_MAC = Db.PIF.get_MAC ~__context ~self:primary_slave_ref in
+              if rc.pIF_MAC <> primary_slave_MAC then {rc with API.pIF_MAC = primary_slave_MAC} else rc
```

Comment:
We still need to update the MAC in the DB, as well as updating the local record that we pass around in this function.